src/cpuinfo_x86.c

```diff
@@ -1330,6 +1330,44 @@ static void ParseCpuId(const uint32_t max_cpuid_leaf, X86Info* info,
     features->ssse3 = GetDarwinSysCtlByName("hw.optional.supplementalsse3");
     features->sse4_1 = GetDarwinSysCtlByName("hw.optional.sse4_1");
     features->sse4_2 = GetDarwinSysCtlByName("hw.optional.sse4_2");
+#elif defined(CPU_FEATURES_OS_FREEBSD)
+    // Handling FreeBSD platform through parsing /var/run/dmesg.boot.
+    const int fd = CpuFeatures_OpenFile("/var/run/dmesg.boot");
+    if (fd >= 0) {
+      StackLineReader reader;
+      StackLineReader_Initialize(&reader, fd);
+      for (;;) {
+        const LineResult result = StackLineReader_NextLine(&reader);
+        if (CpuFeatures_StringView_StartsWith(result.line,
+                                              str("  Features=")) ||
+            CpuFeatures_StringView_StartsWith(result.line,
+                                              str("  Features2="))) {
+          // Lines of interests are of the following form:
+          // "  Features=0x1783fbff<PSE36,MMX,FXSR,SSE,SSE2,HTT>"
+          // We replace '<', '>' and ',' with space so we can search by
+          // whitespace separated word.
+          for (size_t i = 0; i < result.line.size; ++i) {
+            if (result.line[i] == '<' || result.line[i] == '>' ||
+                result.line[i] == ',')
+              result.line[i] == ' ';
+          }
+          if (CpuFeatures_StringView_HasWord(value, "SSE"))
+            features->sse = true;
```

Comment:
why not using:
features->sse = CpuFeatures_StringView_HasWord(value, "SSE");
like in android just below ?

Reply:
I'm grepping two lines, the ones with `Features=` and `Features2=`.
The features might be on either ones so I changed the logic and enable the bit only if it is present.

Alternatively I could do:
```
if (CpuFeatures_StringView_StartsWith(result.line, str("  Features=")) {
 features->sse = CpuFeatures_StringView_HasWord(value, "SSE"); 
 features->sse2 = CpuFeatures_StringView_HasWord(value, "SSE2"); 
}
if (CpuFeatures_StringView_StartsWith(result.line, str("  Features2=")) {
 features->sse3 = CpuFeatures_StringView_HasWord(value, "SSE3");
 // ...
}
```